### PR TITLE
Sensitive output-variables

### PR DESCRIPTION
**Pull request by MJRichardson**
Support for adding sensitive output-variables: https://github.com/OctopusDeploy/Issues/issues/4541

For PowerShell, ScriptCS, and Bash, I added a new optional parameter to the respective `Set-OctoputVariable` commands.

For F# I added a new function `setSensitiveVariable`.  This is because from what I can see there is no way to add a parameter to an existing F# function in a backwards-compatible way.  And there is no operator overloading.  I don't hate the new function anyway.

### CODE REVIEW

File: source/Calamari/Integration/Scripting/Bash/Bootstrap.sh [hunk 1]
```diff
@@ -89,6 +89,11 @@ function set_octopusvariable
 	then
 		MESSAGE="$MESSAGE value='$(encode_servicemessagevalue "$2")'"
 	fi
+	
+	if [ "$3" = "-sensitive" ]
```
Comment (michaelnoonan) [inline source/Calamari/Integration/Scripting/Bash/Bootstrap.sh]: Bash just makes me feel all warm and fuzzy... I mean drunk.

File: source/Calamari/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1 [hunk 2]
```diff
@@ -139,12 +139,17 @@ function Convert-ServiceMessageValue([string]$value)
 	return [Convert]::ToBase64String($valueBytes)
 }
 
-function Set-OctopusVariable([string]$name, [string]$value) 
+function Set-OctopusVariable([string]$name, [string]$value, [switch]$sensitive) 
 {
 	$name = Convert-ServiceMessageValue($name)
 	$value = Convert-ServiceMessageValue($value)
+	$trueEncoded = Convert-ServiceMessageValue("True")
```
Comment (michaelnoonan) [inline source/Calamari/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1]: `$true = ...` would have been interesting! :)

File: source/Calamari.Tests/Fixtures/Bash/BashFixture.cs [hunk 3]
```diff
@@ -21,6 +21,19 @@ public void ShouldPrintEncodedVariable()
             output.AssertSuccess();
             output.AssertOutput("##octopus[setVariable name='U3VwZXI=' value='TWFyaW8gQnJvcw==']");
         }
+        
+        [Test]
+        [Category(TestEnvironment.CompatibleOS.Nix)]
```
Comment (matt-richardson) [inline source/Calamari.Tests/Fixtures/Bash/BashFixture.cs]: We really need to start supporting WSL...

File: source/Calamari.Tests/Fixtures/PowerShell/PowerShellFixture.cs [hunk 4]
```diff
@@ -249,6 +249,21 @@ public void ShouldSetVariables()
             Assert.AreEqual("World!", variables.Get("TestA"));
         }
 
+        [Test]
+        [Category(TestEnvironment.CompatibleOS.Windows)]
```
Comment (matt-richardson) [inline source/Calamari.Tests/Fixtures/PowerShell/PowerShellFixture.cs]: We really need to start supporting powershell on mac and linux too

File: source/Calamari/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1 [hunk 5]
```diff
@@ -139,12 +139,17 @@ function Convert-ServiceMessageValue([string]$value)
...
 
-	Write-Host "##octopus[setVariable name='$($name)' value='$($value)']"
+    If ($sensitive) {
```
Comment (matt-richardson) [inline source/Calamari/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1]: indenting is a bit off here... good old `tab` vs `spaces`....

File: source/Calamari.Tests/Fixtures/Bash/BashFixture.cs [hunk 6]
```diff
@@ -21,6 +21,19 @@ public void ShouldPrintEncodedVariable()
...
+        [Category(TestEnvironment.CompatibleOS.Mac)]
+        public void ShouldPrintSensitiveVariable()
+        {
+            var output = Invoke(Calamari()
+                .Action("run-script")
+                .Argument("script", GetFixtureResouce("Scripts", "print-sensitive-variable.sh")));
+
+            output.AssertSuccess();
+            output.AssertOutput("##octopus[setVariable name='UGFzc3dvcmQ=' value='Y29ycmVjdCBob3JzZSBiYXR0ZXJ5IHN0YXBsZQ==' sensitive='VHJ1ZQ==']");
```
Comment (matt-richardson) [inline source/Calamari.Tests/Fixtures/Bash/BashFixture.cs]: Was well confused about the encoding here, but eventually figured it out...

**Comment by bigbearzhu** [comment]
Great thanks to see progress on this one. Any one could merge this?

**Pull request merged**